tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp

```diff
@@ -223,6 +225,155 @@ tcf_err_t ecall_CreateSignupData(const sgx_target_info_t* inTargetInfo,
     return result;
 }  // ecall_CreateSignupData
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_CreateSignupDataKME(const sgx_target_info_t* inTargetInfo,
```

Comment:
Please move this function to a separate file. These functions needs not be included in the singleton enclave binary. You could create 3 files 
- signup_enclave.cpp
- signup_enclave_kme.cpp, 
- signup_enclave_wpe.cpp, 

I would ideally create a EnclaveSignupData interface or base class and have separate class for Singleton, KME, WPE that implements this interface or derive from base class. Or as a first step you can crate 3 separate files and write "C" style functions